azure-cs-credential-rotation-one-set/AppStack.cs

```diff
@@ -0,0 +1,315 @@
+// Copyright 2016-2020, Pulumi Corporation.  All rights reserved.
+
+using System;
+using Pulumi;
+using Pulumi.AzureNextGen.Authorization.Latest;
+using Pulumi.AzureNextGen.EventGrid.V20200401Preview.Inputs;
+using Pulumi.AzureNextGen.EventGrid.V20200401Preview;
+using Pulumi.AzureNextGen.Insights.Latest;
+using Pulumi.AzureNextGen.KeyVault.Latest;
+using Pulumi.AzureNextGen.KeyVault.Latest.Inputs;
+using Pulumi.AzureNextGen.Resources.Latest;
+using Pulumi.AzureNextGen.Sql.Latest;
+using Pulumi.AzureNextGen.Storage.Latest;
+using Pulumi.AzureNextGen.Web.Latest;
+using Pulumi.AzureNextGen.Web.Latest.Inputs;
+using KeyVault = Pulumi.AzureNextGen.KeyVault.Latest;
+using Storage = Pulumi.AzureNextGen.Storage.Latest;
+
+internal class AppStack : Stack
+{
+    [Output]
+    public Output<string> WebAppEndpoint { get; set; }
+
+    public AppStack()
+    {
+        var config = new Config();
+        var resourceGroupName = config.Require("resourceGroupName");
+        var resourceGroup = Output.Create(GetResourceGroup.InvokeAsync(new GetResourceGroupArgs
```

Comment:
Any objection to creating a new resource group here? We try to make our examples self-contained.

Reply:
No objection, but this is precisely the issue I've been trying to get at. If someone using the arm2pulumi generator wants their stack to be self-contained and use `new ResourceGroup`, like the rest of the examples, then they have to fix all the resource group references, which isn't always a quick task with bigger ARM templates. Switching to `Output.Create(new ResourceGroup)` is an easy fix, but doesn't match the other examples.

Reply:
I used `new ResourceGroup`.

Reply:
> No objection, but this is precisely the issue I've been trying to get at. If someone using the arm2pulumi generator wants their stack to be self-contained and use `new ResourceGroup`, like the rest of the examples, then they have to fix all the resource group references, which isn't always a quick task with bigger ARM templates. Switching to `Output.Create(new ResourceGroup)` is an easy fix, but doesn't match the other examples.

I cut https://github.com/pulumi/arm2pulumi/issues/24 to track this. Let me know your thoughts there.